utils/sl/prereq_tools/base.py

```diff
@@ -345,6 +345,19 @@ def update_submodules(self, subdir):
 
     def get(self, subdir, **kw):
         """Downloads sources from a git repository into subdir"""
+
+        # Now checkout the commit_sha if specified
+        passed_commit_sha = kw.get("commit_sha", None)
+        if passed_commit_sha is None:
+            comp = os.path.basename(subdir)
+            print("""
+*********************** ERROR ************************
+No commit_versions entry in utils/build.config for
+%s. Please specify one to avoid breaking the
+build with random upstream changes.
+*********************** ERROR ************************\n""" % comp)
+            raise DownloadFailure(self.url, subdir)
+
```

Comment:
Per your commit message and the danger (and convenience) of allowing the commit to be _master_, we could, if we really wanted to, allow _master_ locally but disallow it in PRs.  Maybe not worth the effort at this point in time though.

Reply:
Nit picking, but we probably want to allow it in PRs.  Using the CI system to test newer versions of upstream is a legitimate thing to try.  We shouldn't land PRs like that however.

I think python should say you should just read the value here, and trap the exception but this code looks fine.

Reply:
That was actually going to be my position initially but I couldn't think of a way that we could allow it in a PR and prevent it (other than detecting it post-landing, in the landing's build/test run) from landing, but upon reflection, that is quite easy.  We could just add a stage to the CI pipeline that checks for _master_ in `build.config` and contributes a failed test to the tests for the PR.

Reply:
We could, but I don't think it's worthwhile.

Reply:
Using the CI system to test newer versions could be done by simply providing a new build.config with newer hash